examples/remix-cms/app/routes/admin/upload.$.ts

```diff
@@ -1,22 +1,19 @@
 import { json, type ActionArgs } from "@remix-run/cloudflare";
-import { storage } from "superflare";
+import { parseMultipartFormData, storage } from "superflare";
 
 export async function action({ request }: ActionArgs) {
-  /**
-   * This is probably inefficient, but it's the only way to get the `File`
-   * which includes useful information like the file name and type.
-   *
-   * Remix offers (unstable) upload handler APIs which allow you to stream the file body to
-   * the destination _before_ it's fully loaded into memory, whiel exposing the file name and file type.
-   * This would be more efficient, but the APIs aren't fully developed, and I couldn't figure
-   * out how to convert them to a proper `File` object to pass to `storage().put()`. We can
-   * revisit this later.
-   */
-  const formData = await request.formData();
-  const file = formData.get("file") as File;
-  const object = await storage().put(file);
+  const formData = await parseMultipartFormData(
+    request,
+    async ({ stream, filename }) => {
+      const object = await storage().putRandom(stream, {
```

Comment:
Hmmm, so if I wanted to have a max file size, I'd need to continuously read the stream, counting the bytes, before handing it off (or another stream) to storage?

Reply:
Right, or manually read the `data` iterator yourself. Kinda tedious.